service/src/main/java/gov/nasa/pds/api/engineering/elasticsearch/KVPQueryBuilder.java

```diff
@@ -0,0 +1,140 @@
+package gov.nasa.pds.api.engineering.elasticsearch;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import gov.nasa.pds.api.engineering.elasticsearch.business.ProductQueryBuilderUtil;
+
+/**
+ * Builds Elasticsearch queries
+ * @author karpenko
+ */
+public class KVPQueryBuilder
+{
+    private static final Logger log = LoggerFactory.getLogger(KVPQueryBuilder.class);
+
+    private String esIndex;
+    
+    private List<String> fields;
+    private Map<String,List<String>> kvps;
+    
+    private boolean filterByArchiveStatus = false;
```

Comment:
Could have easily added filterByArchiveStatus to the bean RequestAndResponseContext rather than moving all the functionality around and adding yet another bean. Concentrating all of the control context (fields, accept content, etc) makes it simpler to provide consistent behavior and control changes.

Reply:
Context already had methods with 11 parameters and constructor with 13. It would require parameter #14. I've never seen any APIs with that many parameters, but it seems that Java supports 255 :-)
```
RequestAndResponseContext.buildRequestAndResponseContext(this.objectMapper, this.getBaseURL(), lidvid, start, limit, fields, sort, onlySummary, versionSelector, this.presetCriteria, accept);

new RequestAndResponseContext(om, base, null, null, lidvid, start, limit, new ArrayList<String>(), new ArrayList<String>(), false, ProductVersionSelector.ORIGINAL, preset, output_format);
```